x-pack/plugin/esql/qa/security/src/javaRestTest/java/org/elasticsearch/xpack/esql/EsqlSecurityIT.java

```diff
@@ -65,23 +65,23 @@ public void indexDocuments() throws IOException {
 
     public void testAllowedIndices() throws Exception {
         for (String user : List.of("test-admin", "user1", "user2")) {
-            Response resp = runESQLCommand(user, "from index | stats sum=sum(value)");
+            Response resp = runESQLCommand(user, "from index | stats sum=sum(value) | limit 1000");
```

Comment:
A quick hack here would be to check the inside `runESQLCommand` if the command ends up with a limit, if not otherwise add it.